src/parser.rs

```diff
@@ -851,6 +900,18 @@ impl Parser {
         Ok(values)
     }
 
+    /// Parse either `ALL` or `DISTINCT`. Returns `true` if `DISTINCT` is parsed and results in a
+    /// `ParserError` if both `ALL` and `DISTINCT` are fround.
+    pub fn parse_all_or_distinct(&mut self) -> Result<bool, ParserError> {
```

Comment:
I wasn't sure exactly where to place this so I kind of guessed. If there's a better spot, please let me know.

Reply:
There's no well-defined order of the members; this spot makes sense to me!